props/saber_fett263_buttons.h

```diff
@@ -4285,18 +4317,48 @@ SaberFett263Buttons() : PropBase() {}
 #endif
         return true;
 
-      case EVENTID(BUTTON_POWER, EVENT_SECOND_HELD_MEDIUM, MODE_OFF):
+      case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_OFF):
         if (menu_) return true;
         if (SetMute(true)) {
           unmute_on_deactivation_ = true;
           FastOn();
         }
         return false;
 
-      case EVENTID(BUTTON_POWER, EVENT_THIRD_HELD_MEDIUM, MODE_OFF):
+      case EVENTID(BUTTON_POWER, EVENT_THIRD_CLICK_LONG, MODE_OFF):
+        if (menu_) return true;      
         DoBattery();
         return true;
 
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+      // Rehearsal Mode
+      case EVENTID(BUTTON_POWER, EVENT_THIRD_HELD_LONG, MODE_OFF):
+        if (rehearse_) {
+          sound_library_.SayCancel();
+          rehearse_ = false;
+          return true;
+        }
+        // Check for existing rehearsal and prompt to overwrite or keep via menu
+        if (saved_choreography.clash_rec[0].stance == SavedRehearsal::STANCE_CLASH || saved_choreography.clash_rec[0].stance == SavedRehearsal::STANCE_LOCKUP) {
```

Comment:
I assume this just mean "if choreography is initialized (by recording or loading)".
A helper function in saved_choreography called "isInitialized" or similar would make that much more clear.

Reply:
updated